pkg/client/api/client.go

```diff
@@ -51,17 +53,33 @@ func (c *client) V1() v1.ClientInterface {
 	return v1.NewWithClient(c.restClient)
 }
 
-func (c *client) Migrate(body *api.Migrate) (*modelV1.Dashboard, error) {
+func (c *client) Migrate(body *api.Migrate, useDefaultDatasource bool) (*modelV1.Dashboard, error) {
 	result := &modelV1.Dashboard{}
-	err := c.restClient.Post().
+	request := c.restClient.Post().
 		APIVersion("").
 		Resource("migrate").
-		Body(body).
-		Do().
-		Object(result)
+		Body(body)
+
+	if useDefaultDatasource {
+		request = request.Query(&migrateQuery{values: map[string][]string{"default-datasource": {"true"}}})
+	}
+
+	err := request.Do().Object(result)
 	return result, err
 }
 
+type migrateQuery struct {
+	values map[string][]string
```

Comment:
you could do that instead :

```go

func (c *client) Migrate(body *api.Migrate, useDefaultDatasource bool) (*modelV1.Dashboard, error) {
	result := &modelV1.Dashboard{}
	err := c.restClient.Post().
		APIVersion("").
		Resource("migrate").
		Body(body).
        Query(&migrateQuery{defaultDatasource: useDefaultDatasource}).
		Do().
		Object(result)
	return result, err
}

type migrateQuery struct {
  defaultDatasource bool
}

func (q *migrateQuery) GetValues() url.Values {
  return url.Values{
    "default-datasource": []string{strconv.FormatBool(q.defaultDatasource)}
  }
}
```

Reply:
thank you very much for the suggestion! I have modified in 7f9e4a2ba044d25265cbd80772498d59886ea881